docs/user_guide/02b_query_engine.md

```diff
@@ -1 +1,398 @@
 # Query Engine
+
+## What is the Query Engine?
+
+The **MedRecord Query Engine** enables users to find node and edges' indices stored in the graph structure efficiently. Thanks to an intuitive interface, complex queries can be performed, allowing you to filter of nodes and edges by their properties and relationships. This section introduces the basic concepts of querying MedRecords and explores advanced techniques for working with complex datasets.
+
+## Example dataset
+
+An example dataset can be used using the [`MedRecord`](medmodels.medrecord.medrecord.MedRecord){target="_blank"}'s method [`from_example_dataset`](medmodels.medrecord.medrecord.MedRecord.from_example_dataset){target="_blank"}.
+
+```{literalinclude} scripts/02b_show_dataset.py
+---
+language: python
+lines: 8
+---
+```
+
+This example dataset includes a set of patients, drugs, diagnoses and procedures. For this section, we will only use the patients, drugs and the edges that connect these two groups.
```

Comment:
`show()` is a local function defined in the script `scripts/02b_show_dataset.py` right? I think for the user it looks like `show()` is something they could use anywhere and that is misleading I guess. I suggest to either: 
- make show a general function
- Use `print()` in the user guide 
- Explain how `show()` is defined